timelock-agent/src/main/java/com/palantir/atlasdb/timelock/paxos/PaxosResourcesFactory.java

```diff
@@ -60,10 +61,27 @@ public static PaxosResources create(
             MetricsManager metrics,
             Supplier<PaxosRuntimeConfiguration> paxosRuntime,
             ExecutorService sharedExecutor) {
+        return createWithVersion(install,
+                metrics,
+                paxosRuntime,
+                sharedExecutor,
+                Optional.empty());
+    }
+
+    public static PaxosResources createWithVersion(TimelockPaxosInstallationContext install,
+            MetricsManager metrics,
+            Supplier<PaxosRuntimeConfiguration> paxosRuntime,
+            ExecutorService sharedExecutor,
+            Optional<OrderableSlsVersion> timeLockVersion) {
```

Comment:
I'd suggest we just put the version in the installation context (together with things like the user agent and cluster addresses).

Reply:
👍